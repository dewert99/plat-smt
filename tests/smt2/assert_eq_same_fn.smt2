(declare-sort S)
(declare-fun f (S) Bool)
(declare-const a S)
(declare-const b S)
(declare-const p Bool)
(declare-const q Bool)
(push)
(assert (= (f b) (f a)))
(assert (= p (f a)))
(assert (= q (f a)))
(pop)
(assert (not q))
(assert (f a))
(assert (f b))
(check-sat)
(get-model)