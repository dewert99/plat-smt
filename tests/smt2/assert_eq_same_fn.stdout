sat
(
 (define-fun a () S (as @v3 S))
 (define-fun b () S (as @v4 S))
 (define-fun f ((x!0 S)) Bool
  (ite (= x!0 (as @v3 S)) true
  (ite (= x!0 (as @v4 S)) true
   false)))
 (define-fun p () Bool false)
 (define-fun q () Bool false)
)
